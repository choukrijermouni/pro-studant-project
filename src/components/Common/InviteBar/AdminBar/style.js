import {
  blue
} from '@pro_boa/ui'

export default {
  invites: {
    padding: '15px 15px 0 !important',
    minWidth: 312,
    maxWidth: 312,
    resize: 'none'
  },
  icon: {
    fontSize: 16,
    color: blue[0],
    marginTop: 25,
    marginRight: 25,
    cursor: 'pointer'
  }
}
