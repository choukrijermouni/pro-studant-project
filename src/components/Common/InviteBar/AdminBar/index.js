import {
  Row,
  TypographyElement,
  TextArea,
  Button,
  Icon,
  Icons,
  Col
} from '@pro_boa/ui'
import { useState } from 'react'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'

const useStyle = createUseStyles(style)

export default () => {
  const { invites, icon } = useStyle()
  const [arrowInvite, setArrowInvite] = useState(true)
  return (
    <>
      <Row>
        <Col pos='left' className='col' grid={9}>
          <TypographyElement
            component='h3'
            variant='heading3'
            align='left'
            spacing='20px 0 30px 24px'
          >
            {Text.invite}
          </TypographyElement>
        </Col>
        <Col pos='right' className='col' grid={3}>
          {
            arrowInvite
              ? <Icon iconName={Icons.roundedUp} style={icon} handleIconClick={() => setArrowInvite(!arrowInvite)} />
              : <Icon iconName={Icons.roundedDown} style={icon} handleIconClick={() => setArrowInvite(!arrowInvite)} />
          }
        </Col>
      </Row>
      {
        arrowInvite &&
          <Row className='row' justify='center'>
            <TextArea label='' textareaClassName={invites} id='1' type='text' placeholder={Text.placeholder} />
            <Button marginButton='24px 0 0 0' label='Inviter' width='312px' />
          </Row>
      }
    </>
  )
}
