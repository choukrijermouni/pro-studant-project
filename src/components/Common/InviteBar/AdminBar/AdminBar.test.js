import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import AdminBar from './'
configure({ adapter: new Adapter() })

describe('>>> AdminBar --- Shallow Render AdminBar component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<AdminBar />)
    tree = renderer.create(
      <AdminBar />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
