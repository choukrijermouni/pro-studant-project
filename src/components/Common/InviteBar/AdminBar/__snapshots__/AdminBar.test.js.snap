// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> AdminBar --- Shallow Render AdminBar component +++ render correctly and match snapshot 1`] = `
Array [
  <div
    className="row-0-2-5 row-d2-0-2-9"
  >
    <div
      className="col col-0-2-14 col-d3-0-2-18"
    >
      <p
        className="text-0-2-19 text-d0-0-2-20"
      >
        Inviter un administrateur
      </p>
    </div>
    <div
      className="col col-0-2-24 col-d3-0-2-28"
    >
      <span
        className="icon-roundeddown icon-0-2-2"
        onClick={[Function]}
      />
    </div>
  </div>,
  <div
    className="row row-0-2-31 row-d2-0-2-35"
  >
    <div
      className="inputGroup-0-2-37"
    >
      
      <textarea
        className="invites-0-2-1 input-0-2-38 input-d0-0-2-40"
        disabled={false}
        id="1"
        placeholder="E-mail dâ€™administrateur"
      />
      
    </div>
    <button
      className="button-0-2-41 button-d0-0-2-43"
      onClick={[Function]}
      type="button"
    >
      <p
        className="text-0-2-45 text-d0-0-2-46 text-0-2-42 text-d1-0-2-44"
      >
        Inviter
      </p>
    </button>
  </div>,
]
`;
